### ISSUE TITLE
Implement Telegram Message Scraping

### ISSUE THREAD

**Issue by nolanjacobson** [masa-finance/masa-oracle#312]
**Summary:**
The purpose of this feature request is to implement functionality for scraping messages from specified Telegram channels.

**Motivation:**
Scraping messages from Telegram channels is crucial for client integrations. This feature will enable us to collect valuable data that can be used to run sentiment analysis.

**Detailed Explanation:**
The feature should leverage the Telegram Bot API to retrieve messages from a given public Telegram channel. It must comply with Telegram's API usage policies and rate limits to ensure uninterrupted service and legal compliance.

**Proposed Solution:**
- Utilize the Telegram Bot API, ensuring the bot is added to the channel with the necessary permissions to read messages.
- Design a rate limit handling strategy to avoid exceeding Telegram's API request thresholds.
- Securely store the scraped messages in a structured format within a database or file system, facilitating easy retrieval for analysis or processing.

**Considerations:**
- Ensure the bot has appropriate permissions set by the channel administrators to access the messages.
- Develop comprehensive error handling to cope with potential API changes, network issues, or unexpected service disruptions.
- Prioritize user privacy and adhere to data protection laws, being transparent about the data collection process and its purpose to all channel participants.

**Related issues:**
- https://github.com/masa-finance/masa-oracle/issues/358
- #357 

**Acceptance Criteria:**
- [ ] The scraping feature must consistently and accurately collect both new and historical messages from the specified Telegram channel.
- [ ] The implementation must not violate Telegram's rate limiting constraints or API usage policies.
- [ ] The data storage solution should be secure, organized, and allow for efficient data retrieval for subsequent use cases.
- [ ] compliance when scraping messages on telegram (TBD)
- [ ] We have a way for a user to invite the bot in a Telegram channel to scrape 
- [ ] users should be able to do search queries and sentiment analysis.
- [ ] Update documentation by adding instructions on how to use the Telegram scraper

**Comment by giovaroma** [comment]
Action items after reviewing with Nolan and Lacy:

[ ] @nolanjacobson double on compliance when scraping messages on telegram. 
[ ] TBD how user will invite the bot to be added to Telegram channel. 
[ ] Add users able to do search queries and sentiment analysis.

**Comment by H34D** [comment]
@nolanjacobson can you please create follow up tickets?

**Comment by mudler** [comment]
This is blocked by #357

**Comment by restevens402** [comment]
cc @teslashibe @mudler @nolanjacobson 

Nolan and I did some pair work trying to resolve the client issues from the Telegram library we are using.
The Client is implemented as a singleton and was using a global context with cancel.  After the first call, the client context is always closed which does not allow any additional calls on the client.

We removed the global context and created a new context.Context for each call to ensure that we were not holding onto a context that was previously cancelled. This did not solve the problem. It appears that the library we are using is closing the context and I will dig deeper into the library to understand if there is a way around the behavior we are observing.  It is not common for a library to close a context for you, so perhaps there is some other feature we are not aware of in using this. I will find out.
